### PR TITLE
simplify web request-type determination and allow for pathed urlPathForActions and urlPathForFiles

### PR DESCRIPTION
**Pull request by evantahler**
Allows `api.config.servers.web.urlPathForFiles` and `api.config.servers.web.urlPathForFiles` to be set to sub-routes, like `/namespace/api` and `/namespace/public`

### CODE REVIEW

File: __tests__/utils/utils.js [hunk 1]
```diff
@@ -181,6 +181,32 @@ describe('Utils', () => {
     })
   })
 
+  describe('utils.arrayStartignMatch', () => {
```
Comment (gcoonrod) [inline __tests__/utils/utils.js]: Typo: `utils.arrayStartingMatch`

File: initializers/utils.js [hunk 2]
```diff
@@ -393,6 +393,22 @@ let responses = await api.utils.asyncWaterfall(jobs)
       return filteredParams
     }
 
+    /**
+    Compare the first n elements of an array with another, logner array
+
+    @param {Array} a The first array, the shorter one
+    @param {Array} b The second array, the longer one
+    */
+    api.utils.arrayStartignMatch = (a, b) => {
```
Comment (gcoonrod) [inline initializers/utils.js]: This assumes that the arrays are sorted. Maybe add that assumption to the jsdoc.

File: __tests__/utils/utils.js (hunk 1 above)
Comment (evantahler) [reply __tests__/utils/utils.js]: 👍

File: initializers/utils.js (hunk 2 above)
Comment (evantahler) [reply initializers/utils.js]: Interesting... are arrays not assumed to be sorted?

Comment (evantahler) [reply initializers/utils.js]: eh, either way, comment added in fe52475dfa04653f5cfcfacf69549f48c875bef7 !

**Comment by evantahler** [comment]
@gcoonrod thanks for the feedback!  Waiting on your review

**Comment by gcoonrod** [comment]
@evantahler Looks like we're getting some intermittent test failures.

**Comment by evantahler** [comment]
re-tested and green

**Pull request merged**